cmd/erasure-utils.go

```diff
@@ -40,11 +41,19 @@ func newHashWriters(diskCount int, algo string) []hash.Hash {
 func newHash(algo string) hash.Hash {
 	switch algo {
 	case "blake2b":
-		return blake2b.New512()
+		h, err := blake2b.New512(nil)
+		if err != nil {
+			panic(fmt.Sprintf("impossible situation: %s", err))
```

Comment:
> Why is this returning an error? seems like all hashing New() never returns an error. any specific reason?

The blake2b implementation is based on [RFC 7693](https://tools.ietf.org/html/rfc7693) - blake2b can be used as a MAC, but the key must be between 0 and 64 bytes long - if it's longer, the New* functions will return an non-nil error.
But your right, the code should never return an error as long as no key is passed.

> Also lets not do fmt.Sprintf here, we should be little more descriptive.

Maybe something like: "used blake2b as MAC with an invalid key" or is ignoring the error the better behavior?